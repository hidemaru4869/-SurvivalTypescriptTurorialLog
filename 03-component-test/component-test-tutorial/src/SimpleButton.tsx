import { useState } from "react";
import './App.css';

export const SimpleButton: () => JSX.Element = () => {
    const [state, setState] = useState(false);
    const handleClick = () => {
        setState((prevState) => !prevState);
    };
    return <button className="SimpleButton" onClick={handleClick}>{state ? "ON" : "OFF"}</button>;
};