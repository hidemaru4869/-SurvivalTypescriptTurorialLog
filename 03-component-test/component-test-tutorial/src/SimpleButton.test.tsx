// FILEPATH: /workspaces/typescript/work/tutorials/03-component-test/component-test-tutorial/src/SimpleButton.test.tsx
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SimpleButton } from './SimpleButton';

describe('SimpleButton', () => {
    it('renders with initial state OFF', () => {
        render(<SimpleButton />);
        const buttonElement = screen.getByRole('button');
        expect(buttonElement).toHaveTextContent('OFF');
    });

    it('toggles state to ON when clicked', async () => {
        render(<SimpleButton />);
        const buttonElement = screen.getByRole('button');
        await userEvent.click(buttonElement);
        expect(buttonElement).toHaveTextContent('ON');
    });

    it('toggles state back to OFF when clicked again', async () => {
        render(<SimpleButton />);
        const buttonElement = screen.getByRole('button');
        await userEvent.click(buttonElement);
        expect(buttonElement).toHaveTextContent('ON');
        await userEvent.click(buttonElement);
        expect(buttonElement).toHaveTextContent('OFF');
    });

    test("描画されてすぐはOFFと表示されている", () => {
        const view = render(<SimpleButton />);
        expect(view.container).toMatchSnapshot();
    });
});