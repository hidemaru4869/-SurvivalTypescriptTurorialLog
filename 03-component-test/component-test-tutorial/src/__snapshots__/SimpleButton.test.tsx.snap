// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleButton 描画されてすぐはOFFと表示されている 1`] = `
<div>
  <button
    class="SimpleButton"
  >
    OFF
  </button>
</div>
`;
